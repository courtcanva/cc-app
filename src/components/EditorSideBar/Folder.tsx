import { Box } from "@chakra-ui/react";
import { useDispatch } from "react-redux";
import React, { useEffect, useState } from "react";
import { AreaTileQty, changeCourtType } from "@/store/reducer/areaTileQtySlice";
import { setActiveDesign } from "@/store/reducer/courtSpecDataSlice";
import { mockTileData } from "../MockData/MockTileData";
import { useStoreSelector } from "@/store/hooks";
import { changeTileColor } from "@/store/reducer/tileSlice";

const Folder: React.FC = () => {
  const dispatch = useDispatch();
  const { designsData, activeCourt } = useStoreSelector((state) => state.courtSpecData);

  const [activateDesign, setActivateDesign] = useState<string>(activeCourt.courtId);
  const { designTileList } = useStoreSelector((state) => state.tile.present);

  useEffect(() => {
    if (designsData === undefined) return;
    setActivateDesign(activeCourt.courtId);
  }, [designsData]);

  const handleCourtSelecting = (courtId: string): void => {
    setActivateDesign(courtId);
    dispatch(setActiveDesign(courtId));
    const selectedDesign = designsData.find((item) => item.courtId === courtId);
    if (designTileList === undefined) return;
    const selectedDesignColor = designTileList.find((item) => item.courtId === courtId);
    if (selectedDesignColor === undefined) return;
    for (const tile of selectedDesignColor.tileColor) {
      const selectedColor = tile.color;
      dispatch(changeTileColor({ selectedColor, location: tile.location }));
    }
    const tileQtyOfSelectedCourt = mockTileData.find(
      (item) => item.name === selectedDesign?.courtName
    )?.tileQty as AreaTileQty[];
    dispatch(changeCourtType(tileQtyOfSelectedCourt));
  };

  return (
    <Box paddingLeft="24px" paddingTop="24px" height="100%" className="scrollbox">
      {designsData.map((design) => {
        const { courtId, courtName, designName } = design;
        return (
          <Box
            key={courtId}
            width="230px"
            height="25px"
            background="transparent"
            marginBottom="18px"
            display="flex"
            alignItems="center"
            justifyContent="left"
            cursor="pointer"
            fontSize="14"
            onClick={() => handleCourtSelecting(courtId)}
            data-testid={courtId}
            _hover={{ border: "2px solid #40B484" }}
            opacity={!activateDesign || activateDesign === courtId ? "1" : "0.4"}
          >
            {designName} - {courtName}
          </Box>
        );
      })}
    </Box>
  );
};

export default Folder;
