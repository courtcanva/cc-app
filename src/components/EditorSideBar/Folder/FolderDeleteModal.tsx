import { useStoreSelector } from "@/store/hooks";
import { useDispatch } from "react-redux";
import { switchSideBar, switchLoginModal } from "@/store/reducer/buttonToggleSlice";
import { getDesignsData, setDefaultCourt, defaultCourt } from "@/store/reducer/courtSpecDataSlice";
import { fetchDesignData, useDeleteDesignMutation } from "@/redux/api/designApi";
import { designMapping } from "@/utils/designMapping";
import { getDesignsTileData } from "@/store/reducer/designsTileListSlice";
import { changeDesignNameList } from "@/store/reducer/designNameSlice";
import { FaTrashAlt } from "react-icons/fa";
import {
  IconButton,
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalFooter,
  ModalBody,
  ModalCloseButton,
  useDisclosure,
  Button,
} from "@chakra-ui/react";

const FolderDeleteModal = () => {
  const dispatch = useDispatch();
  const userData = useStoreSelector((state) => state.user);
  const { activeCourt: selectedCourt } = useStoreSelector((state) => state.courtSpecData);
  const useUserId = userData.userId;

  const [deleteDesign] = useDeleteDesignMutation();
  const handleDeleteDesign = async (e: { preventDefault: () => void }) => {
    dispatch(switchSideBar(false));
    e.preventDefault();
    if (useUserId === "") {
      dispatch(switchLoginModal(true));
      return;
    }
    if (selectedCourt.courtId === "") return;
    await deleteDesign(selectedCourt.courtId);
    dispatch(setDefaultCourt(defaultCourt));
    const design = await fetchDesignData(useUserId);
    if (design.data === undefined) return;
    const { mappedDesignsData, mappedTileData, mappedNameList } = designMapping(design.data);
    dispatch(getDesignsData(mappedDesignsData));
    dispatch(getDesignsTileData(mappedTileData));
    dispatch(changeDesignNameList(mappedNameList));
  };

  const { isOpen, onOpen, onClose } = useDisclosure();
  return (
    <>
      <IconButton
        aria-label="Delete Button"
        colorScheme="transparent"
        icon={
          <FaTrashAlt
            size={12}
            style={{ color: "white", marginTop: "-18px", marginLeft: "-17px", border: "none" }}
          />
        }
        // variant="editorFooterIconBtn"
        onClick={onOpen}
      />

      <Modal isOpen={isOpen} onClose={onClose} isCentered>
        <ModalOverlay />
        <ModalContent>
          <ModalHeader display={"flex"} flexDirection="column" gap="10px" alignItems={"center"}>
            <FaTrashAlt
              size={35}
              style={{ color: "#9F100F", marginTop: "25px", marginBottom: "10px" }}
            />
          </ModalHeader>
          <ModalCloseButton />
          <ModalBody textAlign={"center"}>
            <p>Are you sure you want to remove this design from the folder?</p>
          </ModalBody>
          <ModalFooter display={"flex"} gap="10px" marginTop={"10px"}>
            <Button variant="ghost" onClick={onClose}>
              Cancel
            </Button>
            <Button variant="folderDeleteBtn" onClick={handleDeleteDesign}>
              Delete
            </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </>
  );
};

export default FolderDeleteModal;
