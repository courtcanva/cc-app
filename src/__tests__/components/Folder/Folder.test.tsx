import Folder from "@/components/EditorSideBar/Folder";
import renderWithMockedProvider from "../../utils";
import { screen } from "@testing-library/react";

describe("Render folder", () => {
  it("should render initial empty folder correctly", () => {
    const { container } = renderWithMockedProvider(<Folder />);
    expect(container).toMatchSnapshot();
    const emptyList = screen.getByText(/the folder list is empty/i);
    expect(emptyList).toBeInTheDocument();
  });
});
