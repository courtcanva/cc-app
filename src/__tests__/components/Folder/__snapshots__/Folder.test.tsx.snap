// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render folder should render initial empty folder correctly 1`] = `
<div>
  <div
    class="scrollbox css-10klw3m"
  >
    <p
      class="chakra-text css-0"
    >
      The folder list is empty
    </p>
  </div>
</div>
`;
